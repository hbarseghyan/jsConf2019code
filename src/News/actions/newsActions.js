import { getNews } from "../services/newsService";

export const actionTypes = {
  GET_NEWS: "GET_NEWS",
};

export const updateNews = payload => ({
  type: actionTypes.GET_NEWS,
  payload,
});

export const loadNews = () => async dispatch => {
  const res = await getNews();
  dispatch(updateNews(res));
};
