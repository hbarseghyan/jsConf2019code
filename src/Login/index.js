import React, { useState } from "react";
import { login } from "./actions/loginActions";

const Login = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const handleSubmit = () => {
    login(email, password);
  };

  const handleChange = e => {
    const { value } = e.currentTarget;
    const { fieldName } = e.currentTarget.dataset;
    switch (fieldName) {
      case "email":
        setEmail(value);
        break;
      case "password":
        setPassword(value);
        break;
      default:
        break;
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        onChange={handleChange}
        placeholder="email"
        value={email}
      />
      <input
        data-field-name="password"
        type="text"
        onChange={handleChange}
        placeholder="pa"
        value={password}
      />
      <button type="submit">Log in</button>
    </form>
  );
};

export default Login;
