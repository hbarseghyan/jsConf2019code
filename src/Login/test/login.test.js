import React from "react";
import { shallow } from "enzyme";
import Login from "..";
import * as loginActions from "../actions/loginActions";

describe("Login", () => {
  let login;
  beforeEach(() => {
    jest.unmock("../actions/loginActions");

    loginActions.login = jest.fn();

    login = shallow(<Login />);
    login.find("form").simulate("submit", {});
  });

  it("renders properly", () => {
    expect(login).toMatchSnapshot();
  });

  it("call login", () => {
    expect(loginActions.login).toHaveBeenCalled();
  });
});
