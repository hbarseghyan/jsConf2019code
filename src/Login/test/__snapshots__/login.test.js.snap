// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders properly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="email"
    type="text"
    value=""
  />
  <input
    data-field-name="password"
    onChange={[Function]}
    placeholder="pa"
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Log in
  </button>
</form>
`;
